tests/test_sattle.py

```diff
@@ -1,76 +1,391 @@
+
+
+    def test_extend_line_horizontal(self):
+        """Test that the horizontal lines are handled correctly"""
+
+        x1, y1, x2, y2, length = np.array([0.0]), np.array([10.0]), np.array([10.0]), np.array([10.0]), 15.0
+        expected_x1, expected_y1, expected_x2, expected_y2 = -15.0, 10.0, 25.0, 10.0
+        result = SattleFilterTask._extend_line(x1, y1, x2, y2, length)
+        self.assertAlmostEqual(result[0].item(), expected_x1, places=3)
+        self.assertAlmostEqual(result[1].item(), expected_y1, places=3)
+        self.assertAlmostEqual(result[2].item(), expected_x2, places=3)
+        self.assertAlmostEqual(result[3].item(), expected_y2, places=3)
+
+        x1, y1, x2, y2, length = np.array([10.0]), np.array([10.0]), np.array([0.0]), np.array([10.0]), 15.0
+        expected_x1, expected_y1, expected_x2, expected_y2 = 25.0, 10.0, -15.0, 10.0
+        result = SattleFilterTask._extend_line(x1, y1, x2, y2, length)
+        self.assertAlmostEqual(result[0].item(), expected_x1, places=3)
+        self.assertAlmostEqual(result[1].item(), expected_y1, places=3)
+        self.assertAlmostEqual(result[2].item(), expected_x2, places=3)
+        self.assertAlmostEqual(result[3].item(), expected_y2, places=3)
+
+    def test_extend_line_vertical(self):
+        """Test that the vertical lines are handled correctly"""
+        x1, y1, x2, y2, length = np.array([0.0]), np.array([0.0]), np.array([0.0]), np.array([10.0]), 15.0
+        expected_x1, expected_y1, expected_x2, expected_y2 = 0.0, -15.0, 0.0, 25.0
+        result = SattleFilterTask._extend_line(x1, y1, x2, y2, length)
+        self.assertAlmostEqual(result[0].item(), expected_x1, places=3)
+        self.assertAlmostEqual(result[1].item(), expected_y1, places=3)
+        self.assertAlmostEqual(result[2].item(), expected_x2, places=3)
+        self.assertAlmostEqual(result[3].item(), expected_y2, places=3)
+
+        x1, y1, x2, y2, length = np.array([0.0]), np.array([10.0]), np.array([0.0]), np.array([0.0]), 15.0
+        expected_x1, expected_y1, expected_x2, expected_y2 = 0.0, 25.0, 0.0, -15.0
+        result = SattleFilterTask._extend_line(x1, y1, x2, y2, length)
+        self.assertAlmostEqual(result[0].item(), expected_x1, places=3)
+        self.assertAlmostEqual(result[1].item(), expected_y1, places=3)
+        self.assertAlmostEqual(result[2].item(), expected_x2, places=3)
+        self.assertAlmostEqual(result[3].item(), expected_y2, places=3)
+
+    def test_check_corners(self):
+        """ Test that the lat and lon extremes are handled correctly."""
+        corners = np.array([[20.0, -10.0, 380.0, 10.0, 10.0, 354.0],
+                           [30.0, 20.0, 20.0, -99.0, 99.0, 99.0]])
+        result = SattleFilterTask._check_corners(corners)
+        expected_result = np.array([[20.0, 350, 20, 190, 190, 174.0],
+                           [30.0, 20.0, 20.0, -81, 81.0, 81.0]])
+        np.testing.assert_array_equal(result, expected_result)
 
```

Comment:
is there a test for `_normalize_coordinates`?

Reply:
Ah, thanks for catching that.  `_check_corners` became `_normalize_coordinates` right before I sent it to you and I didn't catch the rename in test_sattle.py. Should have run the unit tests one more time because it did cause a failure!